scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/constraints/resources/StateQueryParam.java

```diff
@@ -37,27 +38,27 @@ public SerializedValue getValue(SimulationResults simulationResults, Plan plan,
       //TODO: improve Interval to allow querying containment in interval directly
       //for now we look for the last matching profile segment, if any (last to get latest if any overlaps)
       final var profile = simulationResults.realProfiles.get(this.resourceName);
-      final Predicate<LinearProfilePiece> containsQueryTimeP = (piece)
-          -> !Interval.intersect(piece.interval, Interval.at(queryT)).isEmpty();
-      final var piece = Lists.reverse(profile.profilePieces)
+      final Predicate<Segment<LinearEquation>> containsQueryTimeP = (piece)
+          -> !Interval.intersect(piece.interval(), Interval.at(queryT)).isEmpty();
+      final var piece = profile.profilePieces
                              .stream()
                              .filter(containsQueryTimeP)
                              .findFirst()
                              .orElseThrow(() -> new Error(
-                                 "Linear profile for %s not have a segment at the desired time %s"
-                                     .formatted(resourceName, queryT)));
-      return SerializedValue.of(piece.valueAt(queryT));
+                                 "Linear profile for %s not have a segment at the desired time %s".formatted(resourceName, queryT))
+                             );
+      return SerializedValue.of(piece.value().valueAt(queryT));
     } else if (simulationResults.discreteProfiles.containsKey(this.resourceName)) {
       //TODO: improve the profile data structure to allow fast time-keyed query
       //TODO: improve Interval to allow querying containment in interval directly
       //for now we look for the last matching profile segment, if any (last to get latest if any overlaps)
       final var profile = simulationResults.discreteProfiles.get(this.resourceName);
-      final Predicate<DiscreteProfilePiece> containsQueryTimeP = (piece)
-          -> !Interval.intersect(piece.interval, Interval.at(queryT)).isEmpty();
-      final var matchPiece = Lists.reverse(profile.profilePieces).stream()
+      final Predicate<Segment<SerializedValue>> containsQueryTimeP = (piece)
+          -> !Interval.intersect(piece.interval(), Interval.at(queryT)).isEmpty();
+      final var matchPiece = profile.profilePieces.stream()
```

Comment:
Just trying to understand - is the reason this reverse is unnecessary because there can never be overlaps? (And there never were overlaps before, right, so this code was overly defensive about that?).

There's a comment a few lines up that becomes stale after this change - I suggest we just delete it to reduce future confusion. (The one about "for now we look for the last matching profile segment, if any")

Reply:
Yes, I think the reverse was there in case of overlaps. I'll edit the comment

Reply:
updated comment, also improved the predicate. (Interval's interface has improved since this was written.)